dali/test/python/test_pipeline.py

```diff
@@ -1654,3 +1654,132 @@ def test_invoke_serialize_error_handling_string():
 @raises(TypeError, "*define_graph*callable*")
 def test_invoke_serialize_error_handling_not_string():
     _identity_pipe().serialize(42)
+
+
+def check_dtype_ndim(dali_pipeline, output_dtype, output_ndim, n_outputs):
+    def ndim_dtype_matches(test_value, ref_value):
+        ref_value = ref_value if isinstance(ref_value, (list, tuple)) else [ref_value] * n_outputs
+        if len(ref_value) != len(test_value):
+            return False
+        for tv, ref in zip(test_value, ref_value):
+            if tv != ref:
+                return False
+        return True
```

Comment:
Just crossed my mind, maybe not the most efficient but more "declarative". 

```suggestion
        return list(ref_value) == list(test_value)
```

Reply:
Yep, that's prolly way better ;)